vNext/src/Orleans/Shims/AppDomain.cs

```diff
@@ -41,16 +39,9 @@ public Assembly[] GetAssemblies()
                 typeof(Exception).GetTypeInfo().Assembly,
                 typeof(AssemblyProcessor).GetTypeInfo().Assembly,
                 // load up classes for testing, until Assembly scanning is supported in vNext
-                typeof(ClassWithCustomCopier).GetTypeInfo().Assembly,
-                typeof(ClassWithCustomSerializer).GetTypeInfo().Assembly,
-                typeof(SerializerTestClass1).GetTypeInfo().Assembly,
-                typeof(SerializerTestClass2).GetTypeInfo().Assembly,
-                typeof(SerializerTestClass3).GetTypeInfo().Assembly,
-                typeof(SerializerTestClass4).GetTypeInfo().Assembly,
-                typeof(SerializerTestClass5).GetTypeInfo().Assembly,
-                typeof(AsyncObserverArg).GetTypeInfo().Assembly,
-                typeof(StreamSubscriptionHandleArg).GetTypeInfo().Assembly,
-                typeof(GenericArg).GetTypeInfo().Assembly
+                Assembly.Load(new AssemblyName("Orleans.NonSiloTests")),
+                Assembly.Load(new AssemblyName("TestInternalGrains")),
+                Assembly.Load(new AssemblyName("TestGrainInterfaces")) 
```

Comment:
Yes, that will work too and more flexible for tests to load only Assemblies they are interested in. 

But since this is a hack, we would end up using `SetAssemblies `method to many tests, while we shouldn't ideally. So why not gather all the code change related to this Hack and put it inside AppDomain. So when the hack can be removed, we just fix AppDomain and we are good. IMO